src/main/puzzle/scenario-history.gd

```diff
@@ -17,17 +17,66 @@ scores, but could eventually be used for displaying statistics or calculating re
 key: (String) Scenario name 
 value: (Array) All RankResults for the specified scenario
 """
-var scenario_history := {}
+var scenario_history := {} setget , get_scenario_history
 
 # how many records we can store before we start deleting old ones
 var history_size := 1000
 
 var _rank_calculator = RankCalculator.new()
 
+func get_scenario_history() -> Dictionary:
+	if not loaded_scenario_history:
+		load_scenario_history()
+	return scenario_history
+
+
+"""
+Calculates the best rank a player's received for a specific scenario.
+
+Parameters:
+	'scenario_name': The name of the scenario to evaluate
+	
+	'property': (Optional) The property evaluated to determine which of the player's RankResults was the best, such
+		as 'score' or 'seconds'. Defaults to 'score'.
+"""
+func get_best_rank(scenario_name: String, property: String = "score") -> float:
+	var best_rank_result: RankResult = get_best_result(scenario_name, property)
+	return best_rank_result.get(property + "_rank") if best_rank_result else 999.0
+
+
+"""
+Calculates the best rank a player's received for a specific scenario.
+
+Parameters:
+	'scenario_name': The name of the scenario to evaluate
+	
+	'property': (Optional) The property evaluated to determine which of the player's RankResults was the best, such
+		as 'score' or 'seconds'. Defaults to 'score'.
+"""
+func get_best_result(scenario_name: String, property: String = "score") -> RankResult:
+	if not loaded_scenario_history:
+		load_scenario_history()
+	var best_rank := 999.0
+	var best_performance: RankResult
+	if scenario_history.has(scenario_name):
+		for rank_result in ScenarioHistory.scenario_history[scenario_name]:
```

Comment:
Class shouldn't need to dereference itself; just use `scenario_history[scenario_name]`

Reply:
Class no longer dereferences itself